Day 11: “How HTML changes in ePub” / Robin Whittleton

robinwhittleton opened the pull request:
﻿﻿﻿I’ve included a copy of the article as ePub so that people can dissect it. Eleventy seems to serve it fine locally but it might be worth testing that on the site at launch. The ePub contains your CSS and logo, which I assume is fine. It’s licensed under CC0 but I’ve credited you as the artist and typographer in the ePub’s metadata. Happy to relicense the ePub if necessary.

robinwhittleton committed:
Add "How HTML changes in ePub", minus associated ePub

robinwhittleton committed:
Epub for “How HTML changes in ePub”

matuzo commented on hell/adventcalendar/2025/11/index.md:
```diff
@@ -1,35 +1,119 @@
+ePubs, at their core, use HTML, just like the websites we build every day. Except, well, there’s a big asterisk after that. Let’s dive into the differences.
 
-Some text.
+A few decades ago XML emerged from the pit. XML – an attempt at an extensible standard for expressing marked up data – could be used for documents, data transfer, and a bunch of other things, and people genuinely liked it (or, much like AI today, pretended to for job security). They liked it so much that a concerted effort was started to take the nascent HTML and rebuild it on top of XML. This pr ...
```
Link XHTML. Either point to the spec or a blog post explaining the difference between HTML4 and XHTML.

matuzo commented on hell/adventcalendar/2025/11/index.md:
```diff
@@ -1,35 +1,119 @@
-Some text.
+A few decades ago XML emerged from the pit. XML – an attempt at an extensible standard for expressing marked up data – could be used for documents, data transfer, and a bunch of other things, and people genuinely liked it (or, much like AI today, pretended to for job security). They liked it so much that a concerted effort was started to take the nascent HTML and rebuild it on top of XML. This pr ...
 
+XHTML was broadly a flop for a handful of reasons. First, most people didn’t care about extensibility of their markup language, as browsers broadly didn’t support natively any of the standard options available at the time. Second, XML requires a document to be syntactically correct: any mistake in your XHTML and you were met with the dreaded yellow screen of death. Third, because a document neede ...
```
I'm super confused. Are you talking about XHTML or XHTML 2? The way I see it, XHTML was an XML-based stricter version of the SGML-based HTML but I could still make the same mistakes without any real consequences. I could write malformed XHTML without seeing a yellow screen of death.

> You could serve your XHTML as HTML, but then you got no benefit for no reward, and was broadly pointless.

My knowledge about XHTML and its history is limited, so maybe it's just me, but I also don't understand that sentence.

This needs more clarification.

matuzo commented on hell/adventcalendar/2025/11/index.md:
```diff
@@ -1,35 +1,119 @@
-Some text.
 
-Some text. Some text.
+ePub: the [specification](https://www.w3.org/TR/epub-33/) you always knew you wanted yet never knew how to use. You get to take your knowledge of the web, and use it to produce little self-contained sets of documents that can be freely distributed as a single file ready for reading on extremely low-power devices, and they even reflow to fit any screen.
```
What's epub?

matuzo commented on hell/adventcalendar/2025/11/index.md:
```diff
@@ -1,35 +1,119 @@
+
+1. You need to have valid, syntactically correct XML markup. If you don’t, your e-reader will complain. This means self-closing tags, correct namespaces, XML attributes in the XML namespace (`xml:lang`), and so on.
+2. You can include other XML languages directly into your XHTML by adding namespaces.
+3. We unlock the the `epub` namespace, which adds additional functionality to your ePub in e-readers.
```
You're switching between "you" and "we"

matuzo commented on hell/adventcalendar/2025/11/index.md:
```diff
@@ -1,35 +1,119 @@
+
+As well as these ePub semantic vocabulary, we also have things like the [Z39.98-2012 Structural Semantics Vocabulary](https://www.daisy.org/z3998/2012/vocab/structure/). This extends the base ePub spec, and takes the form of `z3998`-namespaced attribute _values_, which definitely looks odd if you’re not used to it. You can use these to express even more fine-grained semantic values, for example [ ...
+
+## Want to try this all out?
```
I don't know if this is possible, but could briefly summarise the steps needed to create an epub?

1. Setup xy
2. Write XHTML
3. Use tool XY to convert

matuzo reviewed (commented):
Thanks, I learned a lot.

I left you some comments.

robinwhittleton committed:
Fixes as per review

Updated text, tweaked ePub display

robinwhittleton commented:
OK, I think https://github.com/matuzo/HTMHell/pull/219/commits/c40c5def672b5b1f44979fcacc4178ab9dccfd87 addresses your review points. But please let me know if there are any more areas that are unclear 🙂

matuzo commented:
@robinwhittleton Is the post ready, can I merge it?

robinwhittleton commented:
Go for it!
